### PR TITLE
Unit tests for parameters and generaldata.ini, remove limit on number of MC years

### PR DESCRIPTION
**Pull request by payetvin**

### CODE REVIEW

File: src/tests/src/libs/antares/study/parameters/parameters-tests.cpp [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+
+// ==================
+// Tests section
+// ==================
+
+BOOST_AUTO_TEST_SUITE(parameters_tests)
+
+BOOST_AUTO_TEST_CASE(validParam)
+{
+    BOOST_CHECK_EQUAL(0, 0);
```
Comment (flomnes) [inline src/tests/src/libs/antares/study/parameters/parameters-tests.cpp]: 🥇

File: src/tests/src/libs/antares/study/parameters/parameters-tests.cpp [hunk 2]
```diff
@@ -42,9 +55,145 @@ struct Fixture
...
+    BOOST_CHECK_EQUAL(p.nbYears, 100000);
 }
 
 BOOST_AUTO_TEST_SUITE_END()
+
+
+void Fixture::writeValidFile()
+{
+    std::ofstream outfile(path);
+    outfile << "[general]" << std::endl;
```
Comment (flomnes) [inline src/tests/src/libs/antares/study/parameters/parameters-tests.cpp]: Use multiline string ?
Or
```cpp
outfile << "AAA\n"
        << "BBB\n";
```

File: src/tests/src/libs/antares/study/parameters/parameters-tests.cpp [hunk 3]
```diff
@@ -0,0 +1,227 @@
...
+    std::ofstream outfile(path);
+
+    outfile << "[general]" << std::endl
+            << "nbyears = abc" << std::endl
+            << "custom-scenario = 27" << std::endl
+            << "first.weekday = 21" << std::endl
+
+            << "[other preferences]" << std::endl
+            << "renewable-generation-modelling = abc" << std::endl;
+
```
Comment (a-zakir) [inline src/tests/src/libs/antares/study/parameters/parameters-tests.cpp]:
```suggestion
    outfile << R"([general]
             nbyears = abc
             custom-scenario = 27
             first.weekday = 21

             [other preferences]
             renewable-generation-modelling = abc)";

```

File: src/tests/src/libs/antares/study/parameters/parameters-tests.cpp [hunk 4]
```diff
@@ -0,0 +1,227 @@
...
+BOOST_FIXTURE_TEST_CASE(fixBadValue, Fixture)
+{
+    p.nbYears = 100;
+    p.derated = true;
+    p.nbTimeSeriesThermal = 0;
+
+    p.fixBadValues();
+
+    BOOST_CHECK_EQUAL(p.nbYears, 1);
```
Comment (flomnes) [inline src/tests/src/libs/antares/study/parameters/parameters-tests.cpp:89]: I don't understand why this is expected

File: src/tests/src/libs/antares/study/parameters/parameters-tests.cpp [hunk 5]
```diff
@@ -0,0 +1,227 @@
...
+    BOOST_CHECK_EQUAL(p.nbTimeSeriesThermal, 1);
+
+    p.nbYears = 10000000;
+    p.derated = false;
+
+    p.fixBadValues();
+
+    BOOST_CHECK_EQUAL(p.nbYears, 100000);
```
Comment (flomnes) [inline src/tests/src/libs/antares/study/parameters/parameters-tests.cpp]: Export this special value ? If already exported, don't use the literal value

File: src/tests/src/libs/antares/study/parameters/parameters-tests.cpp [hunk 6]
```diff
@@ -0,0 +1,227 @@
...
+}
+
+BOOST_FIXTURE_TEST_CASE(invalidValues, Fixture)
+{
+    writeInvalidFile();
+    p.loadFromFile(path.string(), version, options);
```
Comment (flomnes) [inline src/tests/src/libs/antares/study/parameters/parameters-tests.cpp]: Test return value to see if an error is emitted when encountering an invalid file.

File: src/tests/src/libs/antares/study/parameters/parameters-tests.cpp (hunk 4 above)
Comment (payetvin) [reply src/tests/src/libs/antares/study/parameters/parameters-tests.cpp:89]: derated set the nb of year to 1

**Comment by a-zakir** [comment]
Hey, do you plan to test wheter data in the .ini are valid? If the code behave as it should, for ex if I do unit-commitment-mode = fast and furious?

**Review by flomnes — APPROVED** [review]
This PR show that error handling is poor. But it does what it says : put unit tests on generaldata.ini.

**Pull request merged**